lib/ukfile/include/uk/file/pollqueue.h

```diff
@@ -376,43 +308,22 @@ uk_pollevent _pollq_lock(struct uk_pollq *q, uk_pollevent req,
  *   non-zero if awoken
  */
 static inline
-int _pollq_wait(struct uk_pollq *q, uk_pollevent req, __nsec deadline)
+int _uk_pollq_wait(struct uk_pollq *q, uk_pollevent req, __nsec deadline)
 {
-	struct uk_poll_ticket **tail;
-	struct uk_thread *__current;
-	struct uk_poll_ticket tick;
-	int timeout;
-
 	/* Mark request in waitmask */
 	(void)uk_or(&q->waitmask, req);
-	/* Compete to register */
-
-	__current = uk_thread_current();
-	tick = (struct uk_poll_ticket){
-		.next = NULL,
-		.thread = __current,
-		.mask = req,
-	};
-	tail = uk_exchange_n(&q->waitend, &tick.next);
-	/* tail is ours alone, safe to link in */
-	UK_ASSERT(!*tail); /* Should be a genuine list tail */
-	*tail = &tick;
-
-	/* Block until awoken */
-	uk_thread_block_until(__current, deadline);
-	uk_rwlock_runlock(&q->waitlock);
-	uk_sched_yield();
-	/* Back, wake up, check if timed out & try again */
-	timeout = deadline && ukplat_monotonic_clock() >= deadline;
-	if (timeout)
-		uk_pollq_cancel_ticket(q, &tick);
-	return !timeout;
+	/* Set events as cookie & wait */
+	uk_waitq_set_cookie(req);
```

Comment:
Feels unfortunate how `_uk_pollq_wait` is called in a loop and it reaccesses a per-cpu variable every time (`uk_thread_current` so `%gs:offset` on x86 or `TPIDR_EL` on ARM) rather than caching its pointer and doing updates through that pointer. But I think it's not pondering on this at this time I guess...